### PR TITLE
PP 2672 Check if return url is valid url

### PR DESCRIPTION
**Pull request by SandorArpa**
- Update RequestValidations to check whether the return URL is a valid url
- Make return_url non compulsory
- Add handling of return url. When return url is not present (prototype case) then we add a pre-defined return url
- Add/refactor tests accordingly

### CODE REVIEW

**Comment by kakumara** [comment]
There is a problem with the commits / rebasing. I can see a merge commit in the commits. Please fix it.

File: pom.xml [hunk 1]
```diff
@@ -175,6 +176,18 @@
             <version>3.4.6</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.powermock</groupId>
```
Comment (kakumara) [inline pom.xml]: What's the motive behind using powermock? So far we'ven been okay with using just `mockito`

File: src/main/java/uk/gov/pay/products/config/ProductsConfiguration.java [hunk 2]
```diff
@@ -32,8 +32,12 @@
     private String productsUiPayUrl;
 
     private String productsApiToken;
+
     private String productsApiToken2;
 
+    @NotNull
+    private String productsUiConfirmUrl;
```
Comment (kakumara) [inline src/main/java/uk/gov/pay/products/config/ProductsConfiguration.java]: Are you sure, is this part of this PR?

**Comment by SandorArpa** [comment]
there was a mix up and had to cherry pick into a new one

**Pull request closed without merging**